### ISSUE TITLE
Add native printing support

### ISSUE THREAD

**Issue by rhertzog** [zim-desktop-wiki/zim-desktop-wiki#718]
Basically I'm forwarding https://bugs.launchpad.net/zim/+bug/206168 and https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=343369 over to GitHub.

> Zim does not have printing support. There is a work-around to "print" to a browser, so you can use the browser to do the real printing. This is implemented in the "PrintToBrowser" plugin.

> In order to do printing natively we need to use the Gtk printing interface (introduced in Gtk+ 2.10). This involves rendering the contents of the page to a cairo context. Examples can be found on the internet how to do this for text only using Pango. We need a little bit more advanced implementation that also takes care of images.

NOTE: given the age of the initial report, the technical description of what's required for the Gtk printing interface might no longer be accurate.

**Comment by jaap-karssenberg** [comment]
Yes description is correct, to utilize the Gtk native printing interface we need to render zim pages on a cairo canvas.
